dot/core/interface.go

```diff
@@ -66,12 +66,12 @@ type BlockState interface {
 	FreeImportedBlockNotifierChannel(ch chan *types.Block)
 	GetFinalisedNotifierChannel() chan *types.FinalisationInfo
 	FreeFinalisedNotifierChannel(ch chan *types.FinalisationInfo)
-	HighestCommonAncestor(a, b common.Hash) (common.Hash, error)
 	SubChain(start, end common.Hash) ([]common.Hash, error)
 	GetBlockBody(hash common.Hash) (*types.Body, error)
 	HandleRuntimeChanges(newState *rtstorage.TrieState, in runtime.Instance, bHash common.Hash) error
 	GetRuntime(blockHash common.Hash) (instance runtime.Instance, err error)
 	StoreRuntime(common.Hash, runtime.Instance)
+	LowestCommonAncestor(a, b common.Hash) (common.Hash, error)
```

Comment:
what I think is not so clear is what means `lower`? eg:

Given the following chain:
```
                               / -> 4 -> 5 (0xA) -> 6
1 (0xU) -> 2 (0xY) -> 3 (0xZ) ->
                               \ -> 4 (0xB) -> 5 -> 6
```
the common ancestors between hashes `0xA` and `0xB` are hashes `0xZ`, `0xY` and `0xU` but the **lowest** common ancestor is `0xZ` (contains the higher block number and which is more close) or is `0xU` (contains the lowest block number and which is far)?

Reply:
Good point (and thanks for the diagram)!

From how `highestCommonAncestor` was and how the new function is, it would be `0XZ`. I guess maybe `DeepestCommonAncestor` would remove any doubt (since depth is from the root, unlike highest/lowest which can be from both perspective).

Reply:
the function in substarte is `lowest_common_ancestor`.  I'm not sure why we named it `HighestCommonAncestor`.